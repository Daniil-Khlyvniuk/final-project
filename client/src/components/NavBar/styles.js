import { makeStyles } from '@mui/styles'

export const useStyles = makeStyles(() => ({
	appbar: {
		'& .css-17f6boi-MuiToolbar-root, .MuiToolbar-root, .MuiToolbar-gutters, .MuiToolbar-regular, .jss2, .css-15sh0n6': {
			padding: 0
		}
	},
	header: {
		backgroundColor: '#ffffff',
		display: 'flex',
		justifyContent: 'space-between',
		alignItems: 'center',
		flexWrap: 'wrap',
	},
	iconsWrapper: {
		display: 'flex',
		justifyContent: 'space-between',
		alignItems: 'center',
		gap: 30,
		['@media (max-width:750px)']: {
			gap: 20
		},
		['@media (max-width:450px)']: {
			gap: 5
		},
		padding: '10px 10px 10px 0'
	}
}))