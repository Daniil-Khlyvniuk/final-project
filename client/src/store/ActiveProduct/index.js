
export { default as activeProductSelector } from './selectors'
export { default as activeProductOperations } from './operations'
import activeProductReducer from './activeProductSlice'

export default activeProductReducer