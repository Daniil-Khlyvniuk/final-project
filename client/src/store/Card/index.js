export { default as cardSelectors } from './selectors'
export { default as cardOperations } from './operations'
import * as cardActions from './cardSlice'

export default cardActions


