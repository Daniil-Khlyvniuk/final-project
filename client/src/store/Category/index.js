export { default as categorySelectors } from './selectors'
export { default as categoryOperations } from './operations'

import * as categoryActions from './categorySlice'
export default categoryActions