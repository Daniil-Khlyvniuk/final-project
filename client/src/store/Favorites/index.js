export { default as favoritesSelectors } from './selectors'
export { default as favoritesOperations } from './operations'

import * as favoritesActions from './favoritesSlice'
export default favoritesActions