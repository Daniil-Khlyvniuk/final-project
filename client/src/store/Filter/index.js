import reducer from './filterSlice'

export {default as filterSelectors} from './selectors'
export {default as filterOperations} from './operations'

export default reducer