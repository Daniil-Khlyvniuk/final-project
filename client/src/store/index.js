import { configureStore } from '@reduxjs/toolkit'
import productsSlice from './Products/productsSlice'
import linksSlice from './Links/'
import categorySlice from './Category/categorySlice'
import sliderSlice from './Slider'
import modalSlice from './Modal/modalSlice'
import shoppingBagSlice from './ShoppingBag/shoppingBagSlice'
import productSlice from './Product/productSlice'
import filterSlice from './Filter'

import userSlice from './User'

const store = configureStore({
	reducer: {
		slides: sliderSlice,
		products: productsSlice,
		links: linksSlice,
		category: categorySlice,
		modal: modalSlice,
		shoppingBag: shoppingBagSlice,
		user: userSlice,
		product : productSlice,
		filter: filterSlice,

	},
	middleware: (getDefaultMiddleware) =>
		getDefaultMiddleware({
			serializableCheck: {
				// Ignore these action types
				ignoredActions: ['modal/modalToggle', 'user/setToken', 'user/fetchUser/pending','user/fetchUser/fulfilled','filter/setFiltersFromQueryString'],
			},
		}),
})

export default store
