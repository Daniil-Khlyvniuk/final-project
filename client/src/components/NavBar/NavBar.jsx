import React from 'react'
import { AppBar, Container, Toolbar } from '@mui/material'
import NavBarLogo from './NavBarLogo/NavBarLogo'
import { Box } from '@mui/system'
import LoginIcon from './NavBarIcons/LoginIcon/LoginIcon'
import FavoriteIcon from './NavBarIcons/FavotiteIcon/FavoriteIcon'
import CartIcon from './NavBarIcons/CartIcon/CartIcon'
import NavBarSearch from './NavBarSearch/NavBarSearch'
import { useStyles } from './styles'
import Category from '../Category/Category'
import LogoutIconHeader from './NavBarIcons/LogoutIcon/LogoutIcon'
import ProfileIcon from './NavBarIcons/ProfileIcon/ProfileIcon'
import { useLocation } from 'react-router-dom'
import { useSelector } from 'react-redux'
import { userSelectors } from '../../store/User'

const Navbar = () => {
	const classes = useStyles()
	const { pathname } = useLocation()
	const user = useSelector(userSelectors.getData())
	const isLoggedIn = !!user

	return (
		<Box sx={{ borderColor: '#373F41', borderBottom: 1 }}>
			<Container maxWidth="lg" sx={{ minWidth: 320 }}>
				<AppBar className={classes.appbar} position="static" sx={{ boxShadow: 'none' }}>
					<Toolbar className={classes.header}>
						<NavBarLogo />
						{pathname !== '/shop/catalog' && <Category />}
						<NavBarSearch />
						<Box className={classes.iconsWrapper}>
							{!isLoggedIn ? <LoginIcon /> :
								pathname === '/user-profile' ? <LogoutIconHeader /> :
									<ProfileIcon />}
							{isLoggedIn && <FavoriteIcon />}
							<CartIcon />
						</Box>
					</Toolbar>
				</AppBar>
			</Container>
		</Box>
	)
}

export default Navbar