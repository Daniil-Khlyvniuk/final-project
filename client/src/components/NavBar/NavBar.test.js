import React from 'react'
import NavBar from './NavBar'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from '../../store'
import { BrowserRouter } from 'react-router-dom'

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/example/path"
  })
}))

describe('Test NavBar.jsx', () => {
  test('Smoke test NavBar', () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    )
  })

  test('NavBar Logo is rendering', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    )
    getByTestId('navbar-logo')
  })

  test('NavBar Category menu is rendering', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    )
    getByTestId('navbar-category')
  })

  test('NavBar Search input is rendering', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    )
    getByTestId('navbar-search')
  })

  test('NavBar LoginIcon is rendering', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    )
    getByTestId('navbar-login-icon')
  })

  test('NavBar Carticon is rendering', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    )
    getByTestId('navbar-cart-icon')
  })

  test('NavBar Snapshot test', () => {
    const container = render(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })
})