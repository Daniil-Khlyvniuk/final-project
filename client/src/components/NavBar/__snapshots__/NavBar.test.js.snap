// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NavBar.jsx NavBar Snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1szusl3"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-aspcf0-MuiContainer-root"
        >
          <header
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic makeStyles-appbar-43 css-6t7t6l-MuiPaper-root-MuiAppBar-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular makeStyles-header-44 css-1t29gy6-MuiToolbar-root"
            >
              <h1
                class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
                data-testid="navbar-logo"
              >
                <a
                  aria-current="page"
                  class="active"
                  href="/"
                >
                  <div
                    class="makeStyles-logoWrapper-46"
                  >
                    <img
                      alt="logo"
                      src="/logo/header-logo.png"
                    />
                  </div>
                </a>
              </h1>
              <div
                class="MuiBox-root css-0"
                data-testid="navbar-category"
                style="z-index: 1000;"
              >
                <li
                  aria-controls="false"
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-ev1yrh-MuiButtonBase-root-MuiMenuItem-root"
                  id="Category-button"
                  role="menuitem"
                  tabindex="-1"
                >
                  Category
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </li>
              </div>
              <div
                class="MuiBox-root css-1vwk6q9"
                data-testid="navbar-search"
              >
                <div
                  class="css-79elbk"
                >
                  <div
                    class="css-8x9qlq"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-navbarLink-47 css-i4bv87-MuiSvgIcon-root"
                      data-testid="SearchIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </div>
                  <div
                    aria-expanded="false"
                    class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-oyi2zz-MuiAutocomplete-root"
                    role="combobox"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-z5e0z9-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="false"
                        for="combo-box-demo"
                        id="combo-box-demo-label"
                      >
                        Search...
                      </label>
                      <div
                        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-ghsjzk-MuiInputBase-root-MuiInput-root"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-invalid="false"
                          autocapitalize="none"
                          autocomplete="off"
                          class="MuiInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nz481w-MuiInputBase-input-MuiInput-input"
                          id="combo-box-demo"
                          spellcheck="false"
                          type="text"
                          value=""
                        />
                        <div
                          class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                        >
                          <button
                            aria-label="Open"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                            tabindex="-1"
                            title="Open"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                              data-testid="ArrowDropDownIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M7 10l5 5 5-5z"
                              />
                            </svg>
                            <span
                              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="makeStyles-iconsWrapper-45 MuiBox-root css-0"
              >
                <button
                  aria-label="login"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="navbar-login-icon"
                  tabindex="0"
                  title="Login"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-navbarLink-48 css-i4bv87-MuiSvgIcon-root"
                    data-testid="PersonOutlineIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-label="cart"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="navbar-cart-icon"
                  tabindex="0"
                  title="Cart"
                  type="button"
                >
                  <span
                    class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                  >
                    <div
                      class="makeStyles-navbarLink-49 MuiBox-root css-0"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="LocalMallIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 6h-2c0-2.76-2.24-5-5-5S7 3.24 7 6H5c-1.1 0-1.99.9-1.99 2L3 20c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-7-3c1.66 0 3 1.34 3 3H9c0-1.66 1.34-3 3-3zm0 10c-2.76 0-5-2.24-5-5h2c0 1.66 1.34 3 3 3s3-1.34 3-3h2c0 2.76-2.24 5-5 5z"
                        />
                      </svg>
                    </div>
                    <span
                      class="MuiBadge-colorSuccess MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRightRectangular MuiBadge-invisible css-1f5xa6h-MuiBadge-badge"
                    />
                  </span>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </header>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1szusl3"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-aspcf0-MuiContainer-root"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic makeStyles-appbar-43 css-6t7t6l-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular makeStyles-header-44 css-1t29gy6-MuiToolbar-root"
          >
            <h1
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              data-testid="navbar-logo"
            >
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                <div
                  class="makeStyles-logoWrapper-46"
                >
                  <img
                    alt="logo"
                    src="/logo/header-logo.png"
                  />
                </div>
              </a>
            </h1>
            <div
              class="MuiBox-root css-0"
              data-testid="navbar-category"
              style="z-index: 1000;"
            >
              <li
                aria-controls="false"
                aria-expanded="false"
                aria-haspopup="true"
                class="MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root css-ev1yrh-MuiButtonBase-root-MuiMenuItem-root"
                id="Category-button"
                role="menuitem"
                tabindex="-1"
              >
                Category
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </li>
            </div>
            <div
              class="MuiBox-root css-1vwk6q9"
              data-testid="navbar-search"
            >
              <div
                class="css-79elbk"
              >
                <div
                  class="css-8x9qlq"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-navbarLink-47 css-i4bv87-MuiSvgIcon-root"
                    data-testid="SearchIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                    />
                  </svg>
                </div>
                <div
                  aria-expanded="false"
                  class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-oyi2zz-MuiAutocomplete-root"
                  role="combobox"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary css-z5e0z9-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="combo-box-demo"
                      id="combo-box-demo-label"
                    >
                      Search...
                    </label>
                    <div
                      class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-ghsjzk-MuiInputBase-root-MuiInput-root"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-invalid="false"
                        autocapitalize="none"
                        autocomplete="off"
                        class="MuiInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nz481w-MuiInputBase-input-MuiInput-input"
                        id="combo-box-demo"
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                      >
                        <button
                          aria-label="Open"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                          tabindex="-1"
                          title="Open"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="ArrowDropDownIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M7 10l5 5 5-5z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="makeStyles-iconsWrapper-45 MuiBox-root css-0"
            >
              <button
                aria-label="login"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
                data-testid="navbar-login-icon"
                tabindex="0"
                title="Login"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium makeStyles-navbarLink-48 css-i4bv87-MuiSvgIcon-root"
                  data-testid="PersonOutlineIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 5.9c1.16 0 2.1.94 2.1 2.1s-.94 2.1-2.1 2.1S9.9 9.16 9.9 8s.94-2.1 2.1-2.1m0 9c2.97 0 6.1 1.46 6.1 2.1v1.1H5.9V17c0-.64 3.13-2.1 6.1-2.1M12 4C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 9c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="cart"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
                data-testid="navbar-cart-icon"
                tabindex="0"
                title="Cart"
                type="button"
              >
                <span
                  class="MuiBadge-root css-1c32n2y-MuiBadge-root"
                >
                  <div
                    class="makeStyles-navbarLink-49 MuiBox-root css-0"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="LocalMallIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 6h-2c0-2.76-2.24-5-5-5S7 3.24 7 6H5c-1.1 0-1.99.9-1.99 2L3 20c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2zm-7-3c1.66 0 3 1.34 3 3H9c0-1.66 1.34-3 3-3zm0 10c-2.76 0-5-2.24-5-5h2c0 1.66 1.34 3 3 3s3-1.34 3-3h2c0 2.76-2.24 5-5 5z"
                      />
                    </svg>
                  </div>
                  <span
                    class="MuiBadge-colorSuccess MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRightRectangular MuiBadge-invisible css-1f5xa6h-MuiBadge-badge"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </header>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
